src/MediaWiki.ts

```diff
@@ -257,7 +296,7 @@ class MediaWiki {
     const entries = body.query.general;
 
     // Base will contain the default encoded article id for the wiki.
-    const mainPage = decodeURIComponent(entries.base.split('/').pop());
+    const mainPage = this.getMainPage(entries.base.split('/').pop());
```

Comment:
This is wrong. This is `getMainPage()` which should rely on `getSiteInfo()`.

Reply:
ok